Userland/Services/FileOperation/main.cpp

```diff
@@ -28,61 +29,43 @@ struct WorkItem {
     off_t size;
 };
 
-static int perform_copy(Vector<StringView> const& sources, String const& destination);
-static int perform_move(Vector<StringView> const& sources, String const& destination);
-static int perform_delete(Vector<StringView> const& sources);
-static int execute_work_items(Vector<WorkItem> const& items);
-static void report_error(String message);
-static void report_warning(String message);
+static ErrorOr<int> perform_copy(Vector<StringView> const& sources, String const& destination);
+static ErrorOr<int> perform_move(Vector<StringView> const& sources, String const& destination);
+static ErrorOr<int> perform_delete(Vector<StringView> const& sources);
+static ErrorOr<int> execute_work_items(Vector<WorkItem> const& items);
```

Comment:
These are all directly returned via `serenity_main()` which is has a prototype in LibMain.h requiring `ErrorOr<int>`.

It could be "translated" in `serenity_main()` from `ErrorOr<void>` to `ErrorOr<int>`, but that doesn't seem worth it. What do you think?

Reply:
Ah, I see. That one I had missed before.

The question would then be if we just want those functions to tell us whether they errored or not (and handle things like specific return values fully in `serenity_main()`), or should the helper functions set the return value directly?

I'd favor the first one (i.e. convert to `ErrorOr<void>` and then `TRY` as usual in `serenity_main`), but I'll leave that decision to you (and others who might weigh in).

Reply:
I prefer the latter. I don't see a benefit of duplicated the error handling logic into `serenity_main()`